docs/src/catalyst_applications/advanced_simulations.md

```diff
@@ -181,20 +182,28 @@ sol = solve(oprob; callback = ps_cb)
 plot(sol)
 ```
 
-Next, we can also use a callback to change the parameters of a system. The following code
-plots the concentration of a two-state system, as we change the equilibrium
-constant between the two states:
+Next, we can also use a callback to change the parameters of a system. The
+following code plots the concentration of a two-state system, as we change the
+equilibrium constant between the two states. Notice we now need to use `setp` to
+construct a function that can update a parameter within the integrator
 ```@example ex2
 rn = @reaction_network begin
     (k,1), X1 <--> X2
 end
+rn = complete(rn)
 u0 = [:X1 => 10.0, :X2 => 0.0]
 tspan = (0.0, 20.0)
 p = [:k => 1.0]
 oprob = ODEProblem(rn, u0, tspan, p)
 
+# the following code creates a function, `setk` that can
+# change parameters within an integrator
+using ModelingToolkit: setp
+setk = setp(rn, :k)
+
+# now we create our callback and solve
 condition = [5.0]
-affect!(integrator) = integrator[:k] = 5.0
+affect!(integrator) = (setk(integrator, 5.0); nothing)
```

Comment:
Apparently, the bugs introduced by the SII update will be fixed together with the MTK v9 release, which should have been last week, so hopefully (fingers very crossed) it should be soon.

Either way, I think we should use the 
```julia
integrator.ps[:k] = 5.0
```
notation throughout tutorials, rather than
```julia
setk = setp(rn, :k)
setk(integrator, 5.0)
```